> Looks good! I was able to run through all of the patch scenarios successfully. I think I encountered a regression with the `prepare all` command, however:
> 
> ```
> ➜ yes | GBM_WPMOBILE_ORG=derekblank GBM_WORDPRESS_ORG=derekblank go run main.go release prepare all 1.110.0
> [INFO] Preparing Gutenberg for release 1.110.0
> [INFO] Cloning Gutenberg to /var/folders/xk/xmtl5xjx33v_bp94749wm0j00000gn/T/gbm-2609010841/gb
> Cloning into '.'...
> warning: Could not find remote branch  to clone.
> fatal: Remote branch  not found in upstream origin
> [ERROR] error cloning the Gutenberg repository: exit status 128
> ```
> 
> This happens on both the main repos and my forked repos. I think this _might_ be coming from ` build.Base.Ref`. Is `prepare all` still a command, or should we run them individually as `gb` and `gbm`? If we're running them individually, we should remove `all` and update the docs.

Ah thanks for checking that @derekblank ! I forgot to update the all command. I can push a fix.

--- a/cli/cmd/release/prepare/all.go
+++ b/cli/cmd/release/prepare/all.go
@@ -33,13 +33,13 @@ var allCmd = &cobra.Command{
 
 		console.Info("Preparing Gutenberg for release %s", version)
 
-		gbPr, err = release.CreateGbPR(version, gbDir, noTag)
+		gbPr, err = release.CreateGbPR(version.String(), gbDir, noTag)
 		exitIfError(err, 1)
 		console.Info("Finished preparing Gutenberg PR")
 
 		console.Info("Preparing Gutenberg Mobile for release %s", version)
 
-		pr, err := release.CreateGbmPR(version, gbmDir)
+		pr, err := release.CreateGbmPR(version.String(), gbmDir)
 		exitIfError(err, 1)
 		console.Info("Finished preparing Gutenberg Mobile PR")
 

--- a/cli/cmd/release/prepare/gb.go
+++ b/cli/cmd/release/prepare/gb.go
@@ -16,7 +16,7 @@ var gbCmd = &cobra.Command{
 
 		console.Info("Preparing Gutenberg for release %s", version)
 
-		pr, err := release.CreateGbPR(version, tempDir, noTag)
+		pr, err := release.CreateGbPR(version.String(), tempDir, noTag)
 		exitIfError(err, 1)
 
 		console.Info("Created PR %s", pr.Url)

--- a/cli/cmd/release/prepare/gbm.go
+++ b/cli/cmd/release/prepare/gbm.go
@@ -16,7 +16,7 @@ var gbmCmd = &cobra.Command{
 
 		console.Info("Preparing Gutenberg Mobile for release %s", version)
 
-		pr, err := release.CreateGbmPR(version, tempDir)
+		pr, err := release.CreateGbmPR(version.String(), tempDir)
 		exitIfError(err, 1)
 
 		console.Info("Created PR %s", pr.Url)

--- a/cli/cmd/release/prepare/root.go
+++ b/cli/cmd/release/prepare/root.go
@@ -8,12 +8,14 @@ import (
 	wp "github.com/wordpress-mobile/gbm-cli/cmd/workspace"
 	"github.com/wordpress-mobile/gbm-cli/pkg/console"
 	"github.com/wordpress-mobile/gbm-cli/pkg/gbm"
+	"github.com/wordpress-mobile/gbm-cli/pkg/semver"
 )
 
 var exitIfError func(error, int)
 var keepTempDir, noTag bool
 var workspace wp.Workspace
-var tempDir, version string
+var tempDir string
+var version semver.SemVer
 
 var PrepareCmd = &cobra.Command{
 	Use:   "prepare",
@@ -31,9 +33,8 @@ func preflight(args []string) {
 	var err error
 	tempDir = workspace.Dir()
 
-	semver, err := utils.GetVersionArg(args)
+	version, err = utils.GetVersionArg(args)
 	exitIfError(err, 1)
-	version = semver.String()
 
 	// Validate Aztec version
 	if valid := gbm.ValidateAztecVersions(); !valid {

--- a/cli/pkg/release/search.go
+++ b/cli/pkg/release/search.go
@@ -3,7 +3,6 @@ package release
 import (
 	"fmt"
 
-	"github.com/wordpress-mobile/gbm-cli/pkg/console"
 	"github.com/wordpress-mobile/gbm-cli/pkg/gh"
 	"github.com/wordpress-mobile/gbm-cli/pkg/repo"
 )
@@ -39,7 +38,6 @@ func FindAndroidReleasePr(version string) (gh.PullRequest, error) {
 	title := fmt.Sprintf(IntegratePrTitle+" in:title", version)
 
 	filter := gh.BuildRepoFilter(repo.WordPressAndroidRepo, "is:pr", label, title)
-	console.Debug(filter.QueryString)
 
 	return gh.SearchPr(filter)
 }

--- a/cli/pkg/release/utils.go
+++ b/cli/pkg/release/utils.go
@@ -231,7 +231,6 @@ func openInEditor(dir string, files []string) error {
 		files[i] = filepath.Join(dir, f)
 	}
 	cmd := exec.Command(editor, files...)
-	console.Debug(cmd.String())
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr

--- a/cli/pkg/shell/cmds.go
+++ b/cli/pkg/shell/cmds.go
@@ -13,6 +13,7 @@ type CmdProps struct {
 type client struct {
 	cmd       func(...string) error
 	cmdInPath func(string, ...string) error
+	dir       string
 }
 
 func execute(cmd *exec.Cmd, dir string, verbose bool) error {
@@ -34,6 +35,7 @@ func NewNpmCmd(cp CmdProps) NpmCmds {
 			cmd := exec.Command("npm", cmds...)
 			return execute(cmd, path, cp.Verbose)
 		},
+		dir: cp.Dir,
 	}
 }
 
@@ -47,6 +49,7 @@ func NewGitCmd(cp CmdProps) GitCmds {
 			cmd := exec.Command("git", cmds...)
 			return execute(cmd, path, cp.Verbose)
 		},
+		dir: cp.Dir,
 	}
 }
 
@@ -60,6 +63,7 @@ func NewBundlerCmd(cp CmdProps) BundlerCmds {
 			cmd := exec.Command("bundle", cmds...)
 			return execute(cmd, path, cp.Verbose)
 		},
+		dir: cp.Dir,
 	}
 }
 
@@ -73,6 +77,7 @@ func NewRakeCmd(cp CmdProps) RakeCmds {
 			cmd := exec.Command("rake", cmds...)
 			return execute(cmd, path, cp.Verbose)
 		},
+		dir: cp.Dir,
 	}
 }
 

--- a/cli/pkg/shell/git.go
+++ b/cli/pkg/shell/git.go
@@ -2,6 +2,8 @@ package shell
 
 import (
 	"fmt"
+	"os/exec"
+	"strings"
 
 	"github.com/wordpress-mobile/gbm-cli/pkg/console"
 )
@@ -21,6 +23,7 @@ type GitCmds interface {
 	PushTag(string, ...string) error
 	Log(...string) error
 	CherryPick(string) error
+	StatConflicts() ([]string, error)
 }
 
 func (c *client) Clone(args ...string) error {
@@ -103,7 +106,31 @@ func (c *client) Log(args ...string) error {
 	return c.cmd(log...)
 }
 
-func (c *client) CherryPick(commit string) error {
-	pick := append([]string{"cherry-pick"}, commit)
+func (c *client) CherryPick(commitOrContinue string) error {
+	if commitOrContinue == "--continue" {
+		c.cmd("add", "--all")
+	}
+	pick := append([]string{"cherry-pick"}, commitOrContinue)
 	return c.cmd(pick...)
+}
+
+func (c *client) StatConflicts() ([]string, error) {
+	cmd := exec.Command("git", "diff", "--name-only", "--relative", "--diff-filter=U")
+	cmd.Dir = c.dir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return []string{}, err
+	}
+
+	files := strings.Split(string(out), "\n")
+
+	conflicts := []string{}
+	for _, file := range files {
+		if file != "" {
+			conflicts = append(conflicts, file)
+		}
+	}
+
+	return conflicts, nil
 }

